engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/servicetask/ServiceTaskElementActivatedHandler.java

```diff
@@ -36,8 +41,13 @@ protected boolean handleState(final BpmnStepContext<T> context) {
       final WorkflowInstanceRecord value = context.getValue();
       final ExecutableServiceTask serviceTask = context.getElement();
 
-      populateJobFromTask(context, value, serviceTask);
-      context.getCommandWriter().appendNewCommand(JobIntent.CREATE, jobCommand);
+      final Optional<DirectBuffer> optJobType =
+          expressionProcessor.evaluateStringExpression(serviceTask.getType(), context);
+
+      if (optJobType.isPresent()) {
+        populateJobFromTask(context, value, optJobType.get(), serviceTask);
+        context.getCommandWriter().appendNewCommand(JobIntent.CREATE, jobCommand);
+      }
```

Comment:
I personally prefer `ifPresent()`.

```suggestion
      expressionProcessor
        .evaluateStringExpression(serviceTask.getType(), context)
        .ifPresent(jobType -> {
          populateJobFromTask(context, value, jobType, serviceTask);
          context.getCommandWriter().appendNewCommand(JobIntent.CREATE, jobCommand);
        });
```

Reply:
I see what you mean. But soon there will be two expressions 'jobType' and 'retries'. Which looks even worse. If you have a good idea how to chain two optionals, I am all ears.

Reply:
Alright. I get your point :+1: